core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/api/rename.kt

```diff
@@ -15,8 +15,13 @@ import org.jetbrains.kotlinx.dataframe.impl.columns.tree.map
 import org.jetbrains.kotlinx.dataframe.kind
 
 internal fun <T, C> RenameClause<T, C>.renameImpl(newNames: Array<out String>): DataFrame<T> {
-    var i = 0
-    return renameImpl { newNames[i++] }
+    // associate old column names with new ones
+    val selectedColumns = df.getColumnsWithPaths(columns)
+    val oldToNew = newNames.mapIndexed { index, newName ->
+        selectedColumns[index].name to newName
```

Comment:
This can now throw an exception even there are fewer columns selected than names given. If we want to maintain the old behavior, we should limit newNames to the number of selected columns before mapping